src/pint/models/timing_model.py

```diff
@@ -799,6 +799,87 @@ def orbital_phase(self, barytimes, anom="mean", radians=True):
         # return with radian units or return as unitless cycles from 0-1
         return anoms * u.rad if radians else anoms / (2 * np.pi)
 
+    def radial_position(self, barytimes):
+        """Return line-of-sight position at barycentric MJD times.
+
+        Parameters
+        ----------
+        barytimes: Time, TOAs, array-like, or float
+            MJD barycentric time(s). The times to compute the
+            orbital phases.  Needs to be a barycentric time in TDB.
+            If a TOAs instance is passed, the barycentering will happen
+            automatically.  If an astropy Time object is passed, it must
+            be in scale='tdb'.  If an array-like object is passed or
+            a simple float, the time must be in MJD format.
+
+        Raises
+        ------
+        ValueError
+            If an astropy Time object is passed with scale!="tdb".
+
+        Returns
+        -------
+        array
+            The line-of-sight position
```

Comment:
Can we explicitly say this position is with respect to the system barycenter (if that is indeed the case)?

Reply:
done